server/src/main/java/org/apache/iotdb/db/metadata/template/Template.java

```diff
@@ -240,22 +218,25 @@ private void constructTemplateTree(String[] alignedPaths, IMeasurementSchema[] s
 
       measurementNames.add(pathNodes[pathNodes.length - 1]);
     }
-    // find and assign common parent node of aligned measurements to commonPar
-    if (prefix.equals("")) {
-      commonPar = (IEntityMNode) templateRoot;
-    } else {
-      commonPar = (IEntityMNode) constructEntityPath(alignedPaths[0]);
-    }
 
     synchronized (this) {
+      // if not aligned now, it will be set to aligned
       if (!alignedPrefix.contains(prefix)) {
         alignedPrefix.add(prefix);
       }
       for (int i = 0; i <= measurementNames.size() - 1; i++) {
-        leafNode =
-            MeasurementMNode.getMeasurementMNode(
-                commonPar, measurementNames.get(i), schemas[i], "");
-        commonPar.addChild(leafNode);
+        // find the parent and add nodes to template
+        if (prefix.equals("")) {
+          leafNode =
+              MeasurementMNode.getMeasurementMNode(null, measurementNames.get(i), schemas[i], "");
+          directNodes.put(leafNode.getName(), leafNode);
+        } else {
+          commonPar = (IEntityMNode) constructEntityPath(alignedPaths[0]);
+          leafNode =
+              MeasurementMNode.getMeasurementMNode(
+                  commonPar, measurementNames.get(i), schemas[i], "");
+          commonPar.addChild(leafNode);
+        }
```

Comment:
Why set alias "". If there's no alias, it should just be set to null.

Reply:
get it, and all alias set to null now